From 996fd06f9ddc008379ecd2801d9f3d3f650f63dd Mon Sep 17 00:00:00 2001
From: Arian <arian.kulmer@web.de>
Date: Tue, 15 Oct 2024 18:58:34 +0200
Subject: [PATCH 6/7] Read AOSP integer camera_launch_source intent in
 STILL_IMAGE_CAMERA case

MIUI changed the camera_launch_source to be a string, but AOSP provides integers.
---
 smali/d/d/a/b4.smali | 83 ++++++++++++++------------------------------
 1 file changed, 27 insertions(+), 56 deletions(-)

diff --git a/smali/d/d/a/b4.smali b/smali/d/d/a/b4.smali
index bb1870a2f..980cfb364 100644
--- a/smali/d/d/a/b4.smali
+++ b/smali/d/d/a/b4.smali
@@ -1181,27 +1181,21 @@
 
     const-string v0, "com.android.systemui.camera_launch_source"
 
-    invoke-virtual {p0, v0}, Landroid/content/Intent;->getStringExtra(Ljava/lang/String;)Ljava/lang/String;
-
-    move-result-object p0
-
-    const-string v0, "lockscreen_affordance"
-
-    .line 7
-    invoke-static {p0, v0}, Landroid/text/TextUtils;->equals(Ljava/lang/CharSequence;Ljava/lang/CharSequence;)Z
-
-    move-result v0
+    const/4 v1, -0x1
 
-    if-nez v0, :cond_3
+    invoke-virtual {p0, v0, v1}, Landroid/content/Intent;->getIntExtra(Ljava/lang/String;I)I
 
-    const-string v0, "power_double_tap"
+    move-result p0
 
-    .line 8
-    invoke-static {p0, v0}, Landroid/text/TextUtils;->equals(Ljava/lang/CharSequence;Ljava/lang/CharSequence;)Z
+    # CAMERA_LAUNCH_SOURCE_POWER_DOUBLE_TAP = 1
+    const/4 v1, 0x1
+    if-eq p0, v1, :cond_3
 
-    move-result p0
+    # CAMERA_LAUNCH_SOURCE_QUICK_AFFORDANCE = 3
+    const/4 v1, 0x3
+    if-eq p0, v1, :cond_3
 
-    if-eqz p0, :cond_4
+    goto :cond_4
 
     :cond_3
     const/4 v2, 0x1
@@ -2858,9 +2852,11 @@
 
     const-string v1, "com.android.systemui.camera_launch_source"
 
-    invoke-virtual {v0, v1}, Landroid/content/Intent;->getStringExtra(Ljava/lang/String;)Ljava/lang/String;
+    const/4 v2, -0x1
 
-    move-result-object v0
+    invoke-virtual {v0, v1, v2}, Landroid/content/Intent;->getIntExtra(Ljava/lang/String;I)I
+
+    move-result v0
 
     .line 2
     new-instance v1, Ljava/lang/StringBuilder;
@@ -2871,7 +2867,7 @@
 
     invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
 
-    invoke-virtual {v1, v0}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+    invoke-virtual {v1, v0}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
 
     invoke-virtual {v1}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
 
@@ -2894,20 +2890,24 @@
 
     move-result v1
 
-    const-string v3, "power_double_tap"
+    # CAMERA_LAUNCH_SOURCE_POWER_DOUBLE_TAP = 1
+    const/4 v4, 0x1
 
     if-eqz v1, :cond_0
 
     .line 4
-    invoke-static {v0, v3}, Landroid/text/TextUtils;->equals(Ljava/lang/CharSequence;Ljava/lang/CharSequence;)Z
+    const/4 v2, 0x1
 
-    move-result p0
+    if-eq v0, v4, :cond_3
 
-    invoke-static {p0}, Ljava/lang/Boolean;->valueOf(Z)Ljava/lang/Boolean;
+    const/4 v2, 0x0
 
-    move-result-object p0
+    :cond_3
+    invoke-static {v2}, Ljava/lang/Boolean;->valueOf(Z)Ljava/lang/Boolean;
 
-    return-object p0
+    move-result-object v2
+
+    return-object v2
 
     .line 5
     :cond_0
@@ -2924,38 +2924,9 @@
     if-eqz p0, :cond_1
 
     .line 6
-    invoke-static {v0, v3}, Landroid/text/TextUtils;->equals(Ljava/lang/CharSequence;Ljava/lang/CharSequence;)Z
-
-    move-result p0
-
-    if-nez p0, :cond_1
-
-    const-string p0, "double_click_volume_down"
-
-    .line 7
-    invoke-static {v0, p0}, Landroid/text/TextUtils;->equals(Ljava/lang/CharSequence;Ljava/lang/CharSequence;)Z
-
-    move-result p0
-
-    if-nez p0, :cond_1
-
-    const-string p0, "stabilizer"
-
-    .line 8
-    invoke-static {v0, p0}, Landroid/text/TextUtils;->equals(Ljava/lang/CharSequence;Ljava/lang/CharSequence;)Z
-
-    move-result p0
-
-    if-nez p0, :cond_1
-
-    const-string p0, "miwatch"
-
-    .line 9
-    invoke-static {v0, p0}, Landroid/text/TextUtils;->equals(Ljava/lang/CharSequence;Ljava/lang/CharSequence;)Z
-
-    move-result p0
+    if-eq v0, v4, :cond_1
 
-    if-eqz p0, :cond_2
+    goto :cond_2
 
     :cond_1
     const/4 v2, 0x1
-- 
2.39.5

